### PR TITLE
[IMP] mrp: improve dates on MO

### PR DESCRIPTION
**Pull request by std-odoo**
- Now, the field ``date_planned_start`` now reflect the ``date_expected`` of the ``move_raw_ids``.
- The field ``date_planned_finished`` reflect the ``date_expected`` of the ``move_finished_ids``.
- Fix the delay alert on the manufacturing order
- Stop date propagation if the manufacturing order is planned

Task #1970595

### CODE REVIEW

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 import datetime
+from datetime import timedelta
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_production.py]: remove + use datetime.timedelta

File: addons/mrp/models/mrp_production.py [hunk 2]
```diff
@@ -478,12 +490,10 @@ def onchange_picking_type(self):
...
         res = super(MrpProduction, self).write(vals)
-        if 'date_planned_start' in vals:
-            moves = (self.mapped('move_raw_ids') + self.mapped('move_finished_ids')).filtered(
-                lambda r: r.state not in ['done', 'cancel'])
-            moves.write({
-                'date_expected': fields.Datetime.to_datetime(vals['date_planned_start']),
-            })
+
+        if not self.routing_id and vals.get('date_planned_start') and 'date_planned_finished' not in vals:
+            self.date_planned_finished = self.date_planned_start + datetime.timedelta(hours=1)
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_production.py]: you could do it before the call to super and only set the vals (no write inside write)

**Comment by sle-odoo** [comment]
we need to stop the propagation if it encounters a planned mo
the removal of the onchange setting the date planned finished is problematic

File: addons/mrp/models/mrp_production.py [hunk 3]
```diff
@@ -699,6 +710,8 @@ def button_plan(self):
         orders_to_plan = self.filtered(lambda order: order.routing_id and order.state == 'confirmed')
         for order in orders_to_plan:
             order.move_raw_ids.filtered(lambda m: m.state == 'draft')._action_confirm()
+            # stop the propagation after planning
```
Comment (sle-odoo) [inline addons/mrp/models/mrp_production.py]:
``` python
# `propagate_date` enables the automatic rescheduling which could lead to hard to
# undertand behavior if a manufacturing order is planned, ie if the work orders do
# have their leaves booked in the workcenter calendar. We thus disable the
# automatic rescheduling in this scenario.
```

File: addons/mrp/models/stock_move.py [hunk 4]
```diff
@@ -244,6 +244,8 @@ def _get_upstream_documents_and_responsibles(self, visited):
     def _delay_alert_get_documents(self):
         res = super(StockMove, self)._delay_alert_get_documents()
         productions = self.mapped('raw_material_production_id')
+        if not productions:
+            productions = self.mapped('production_id')
```
Comment (sle-odoo) [inline addons/mrp/models/stock_move.py]: productions = self. raw_material_production_id | self.production_id

File: addons/mrp/models/mrp_production.py [hunk 5]
```diff
@@ -488,6 +494,8 @@ def write(self, vals):
                     raise UserError(_('You cannot move a planned manufacturing order.'))
             if 'move_raw_ids' in vals and production.state != 'draft':
                 production.move_raw_ids.filtered(lambda m: m.state == 'draft')._action_confirm()
+            if not production.routing_id and vals.get('date_planned_start') and not vals.get('date_planned_finished'):
```
Comment (amoyaux) [inline addons/mrp/models/mrp_production.py]: I would only do it if the current date_planned_finished is lower than the new date_planned_start

Comment (sle-odoo) [reply addons/mrp/models/mrp_production.py]:
``` python
if not production.routing_id and vals.get('date_planned_start') and not vals.get('date_planned_finished'):
    date_planned_start_dt = fields.Datetime.to_datetime(vals.get('date_planned_start'))
    if production.date_planned_finished < date_planned_start_dt:
        production.date_planned_finished = date_planned_start_dt + datetime.timedelta(hours=1)
```

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Comment by sle-odoo** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**